db/migrate/20190322040535_add_indexes_to_session_activities.rb

```diff
@@ -0,0 +1,10 @@
+class AddIndexesToSessionActivities < ActiveRecord::Migration
+  disable_ddl_transaction!
+
+  def change
+    add_index(:session_activities, :name, unique: false, algorithm: :concurrently)
+    add_index(:session_activities, :status, unique: false, algorithm: :concurrently)
+    add_index(:session_activities, [:id, :originating_request_id], unique: true, algorithm: :concurrently)
```

Comment:
could you explain what you're trying to do with this index?

Reply:
yes, I plan to use this index to do a find / lookup when SAML_Callback returns. So I will set RelayState: { session_activity_id: 1, request_id: 213123123132 }, and I will then use this to do a lookup of the persisted session activity object when the callback returns. Both attributes must be persisted.

You might be asking, why not just use session_activity_id? Well, I think it's possible that a user could intercept the url that is generated and change the session_activity_id before making the request. I want to mitigate against that possibility and ensure that there is a second factor that must match.

Reply:
Ok, in that case I don't think this index is helpful.  id is already indexed as a primary key and originating_request_id typically won't do any additional filtering.  I could see an index on originating_request_id though.

Reply:
really? the way I plan to do almost all the lookups in the callback is SessionActivity.find_by(originating_request_id: , session_activity_id: ) neither of which should ever be null. Why do you think just indexing originating_request_id would be better, i don't think we would every query based off of that column by itself.

Reply:
You're right, it shouldn't make any difference and it will probably only slow down doing INSERTS.

Reply:
made this change.